### PR TITLE
feat: add more configurable parameters in the build image phase

### PR DESCRIPTION
**Pull request by qclc**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -79,19 +79,27 @@ debug: build
...
+#   DOCKER_BUILD_ARGS: additional parameters of the Dockerfile need to be passed in during the image building process,
+#                      e.g.: `--build-arg BUILD_FLAGS=-race`
+#   DOCKERFILE_PATH:   the Dockerfile path used by docker build, default is `kubeadmiral/hack/dockerfiles/Dockerfile`
 #
 # Examples:
 #   # build images with the host arch, it will generate "ghcr.io/kubewharf/kubeadmiral-controller-manager:latest"
 #   make images
 #
-#   # build images with amd64,arm64 arch
+#   # build images with amd64,arm64 arch, and use the mirror source in mainland China during the construction process
```
Comment (SOF3) [inline Makefile:90]: you mean image building process?

File: Makefile [hunk 2]
```diff
@@ -79,19 +79,27 @@ debug: build
...
 #   # note: If you specify multiple architectures, the image tag will be added with the architecture name,e.g.:
 #   # 		"ghcr.io/kubewharf/kubeadmiral-controller-manager:latest-amd64"
 #   # 		"ghcr.io/kubewharf/kubeadmiral-controller-manager:latest-arm64"
-#   make images ARCHS=amd64,arm64
+#   make images ARCHS=amd64,arm64  REGION="cn"
```
Comment (SOF3) [inline Makefile:94]:
```suggestion
#   make images ARCHS=amd64,arm64 REGION="cn"
```

File: Makefile [hunk 3]
```diff
@@ -79,19 +79,27 @@ debug: build
...
+#
+#   # pass parameters to go build and enter debug mode
```
Comment (SOF3) [inline Makefile:96]:
```suggestion
#   # enable race detection in go build
```

File: Makefile [hunk 4]
```diff
@@ -79,19 +79,27 @@ debug: build
...
+#   make images DOCKER_BUILD_ARGS="--build-arg BUILD_FLAGS=-race"
 .PHONY: images
 images:
-	REGISTRY=$(REGISTRY) TAG=$(TAG) ARCHS=$(ARCHS) GOPROXY=$(GOPROXY) REGION=$(REGION) bash hack/make-rules/build-images.sh
+	REGISTRY=$(REGISTRY) TAG=$(TAG) ARCHS=$(ARCHS) GOPROXY=$(GOPROXY) REGION=$(REGION) \
+	DOCKER_BUILD_ARGS="$(DOCKER_BUILD_ARGS)" DOCKERFILE_PATH="$(DOCKERFILE_PATH)" \
```
Comment (SOF3) [inline Makefile:101]: is it legal to further indent this?

**Review by SOF3 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (qclc) [reply Makefile]: yes, the comment has been modified, please take a review.

File: Makefile (hunk 3 above)
Comment (qclc) [reply Makefile]: the note has been modified, please take a review.

**Review by SOF3 — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (SOF3) [reply Makefile]: Actually I meant it looks better to double indent this

File: Makefile [hunk 5]
```diff
@@ -79,19 +79,26 @@ debug: build
...
-#   make images ARCHS=amd64,arm64
+#   make images ARCHS=amd64,arm64 REGION="cn"
+#
+#   # enable race detection in go build
+#   make images DOCKER_BUILD_ARGS="--build-arg BUILD_FLAGS=-race"
 .PHONY: images
 images:
-	REGISTRY=$(REGISTRY) TAG=$(TAG) ARCHS=$(ARCHS) GOPROXY=$(GOPROXY) REGION=$(REGION) bash hack/make-rules/build-images.sh
+	REGISTRY=$(REGISTRY) TAG=$(TAG) ARCHS=$(ARCHS) GOPROXY=$(GOPROXY) REGION=$(REGION) \
+DOCKER_BUILD_ARGS="$(DOCKER_BUILD_ARGS)" DOCKERFILE_PATH="$(DOCKERFILE_PATH)" bash hack/make-rules/build-images.sh
```
Comment (SOF3) [inline Makefile:101]:
```suggestion
	REGISTRY=$(REGISTRY) TAG=$(TAG) ARCHS=$(ARCHS) GOPROXY=$(GOPROXY) REGION=$(REGION) \
		DOCKER_BUILD_ARGS="$(DOCKER_BUILD_ARGS)" DOCKERFILE_PATH="$(DOCKERFILE_PATH)" bash hack/make-rules/build-images.sh
```

**Review by SOF3 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 5 above)
Comment (qclc) [reply Makefile]: The indentation problem has been fixed, please take a review.

**Review by SOF3 — APPROVED** [review]

**Pull request merged**